### PR TITLE
Improve language around `prefect deploy` to not recommend creating a deployment by flow name

### PR DESCRIPTION
**Pull request by AmanSal1**
DESCRIPTION OF THE PULL REQUEST:

Changes have been made in two files: 'test_deploy.py' and 'deploy.py'. Since code deployment through flow names is no longer possible and has been deprecated, the 'prefect deploy --help' command needs to be updated accordingly. Additionally, the 'ValueError' in the 'deploy.py' file has been modified to match the same error in the 'test_deploy.py' file, ensuring consistency and maintaining the integrity of the tests.

Fixes:#9984

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed

### CODE REVIEW

**Comment by AmanSal1** [comment]
@desertaxle Can you review this new PR I guess I have made the required changes now.

**Comment by AmanSal1** [comment]
@desertaxle I guess I have to run black before it can be accepted ?

File: src/prefect/cli/deploy.py [hunk 1]
```diff
@@ -306,13 +306,10 @@ async def _run_single_deploy(
...
-                "An entrypoint or flow name must be provided.\n\nDeploy a flow by"
-                " entrypoint:\n\n\t[yellow]prefect deploy"
-                " path/to/file.py:flow_function[/]\n\nDeploy a flow by"
-                " name:\n\n\t[yellow]prefect project register-flow"
-                " path/to/file.py:flow_function\n\tprefect deploy --flow"
-                " registered-flow-name[/]\n\nYou can also provide an entrypoint or flow"
-                " name in this project's prefect.yaml file."
+                "An entrypoint must be provided:\n\n"
+                " \t[yellow]prefect deploy path/to/file.py:flow_function\n\n"
+                " You can also provide an entrypoint in a prefect.yaml"
```
Comment (serinamarie) [inline src/prefect/cli/deploy.py:311]:
```suggestion
                "You can also provide an entrypoint in a prefect.yaml"
```

**Comment by serinamarie** [comment]
Hi @AmanSal1, thanks so much for the contribution! Could you do me a favor and provide a screenshot of what the `ValueError` looks like now?

File: tests/cli/test_deploy.py [hunk 2]
```diff
@@ -521,7 +521,7 @@ async def test_project_deploy_exits_with_no_name_or_entrypoint_configured(
             invoke_and_assert,
             command="deploy",
             expected_code=1,
-            expected_output_contains="An entrypoint or flow name must be provided.",
+            expected_output_contains="An entrypoint must be provided:"
```
Comment (serinamarie) [inline tests/cli/test_deploy.py:524]:
```suggestion
            expected_output_contains="An entrypoint must be provided:",
```

File: tests/cli/test_deploy.py [hunk 3]
```diff
@@ -1524,7 +1524,7 @@ async def test_project_deploy_exits_with_no_name_or_entrypoint_configured(
             invoke_and_assert,
             command="deploy -n test-name",
             expected_code=1,
-            expected_output_contains="An entrypoint or flow name must be provided.",
+            expected_output_contains="An entrypoint must be provided:"
```
Comment (serinamarie) [inline tests/cli/test_deploy.py:1527]:
```suggestion
            expected_output_contains="An entrypoint must be provided:",
```

Comment (AmanSal1) [reply tests/cli/test_deploy.py:1527]: @serinamarie That's all I have to change ? Will share you the screenshot .

Comment (AmanSal1) [reply tests/cli/test_deploy.py:1527]: @serinamarie This is how it looks now:

Comment (serinamarie) [reply tests/cli/test_deploy.py:1527]: Ah, great. Can you update the description with it, please?

Comment (AmanSal1) [reply tests/cli/test_deploy.py:1527]: "Can you please provide more details? I didn't understand your explanation

Comment (AmanSal1) [reply tests/cli/test_deploy.py:1527]: I guess you are talking about the prefect deploy --description ?

Comment (serinamarie) [reply tests/cli/test_deploy.py:1527]: The pull request description :)

**Comment by serinamarie** [comment]
In addition, would you want to close this?

https://github.com/PrefectHQ/prefect/pull/10093

File: tests/cli/test_deploy.py (hunk 3 above)
Comment (AmanSal1) [reply tests/cli/test_deploy.py]: Done!!

**Comment by AmanSal1** [comment]
@serinamarie Yup I will close it .

File: tests/cli/test_deploy.py (hunk 3 above)
Comment (serinamarie) [reply tests/cli/test_deploy.py]: I'm not seeing it?

Comment (AmanSal1) [reply tests/cli/test_deploy.py]: Could you please tell me why my unit tests are failing? They have been failing in my previous pull requests as well

Comment (serinamarie) [reply tests/cli/test_deploy.py]: They should be resolved if you commit the suggestions I gave 👍

Comment (AmanSal1) [reply tests/cli/test_deploy.py]: In the second commit, I made the code changes according to your suggestion.

**Review by serinamarie — APPROVED** [review]
Thanks for your contribution, @AmanSal1! Look to our [next release 2.10.19 for a shoutout](https://github.com/PrefectHQ/prefect/blob/main/RELEASE-NOTES.md) 🚀

**Comment by AmanSal1** [comment]
@serinamarie Sounds great !!

**Pull request merged**